stackslib/src/chainstate/stacks/boot/pox-4.clar

```diff
@@ -1194,12 +1218,17 @@
 (define-public (delegate-stack-extend
                     (stacker principal)
                     (pox-addr { version: (buff 1), hashbytes: (buff 32) })
-                    (extend-count uint))
+                    (extend-count uint)
+                    (updated-signing-key (optional (buff 33))))
     (let ((stacker-info (stx-account stacker))
           ;; to extend, there must already be an entry in the stacking-state
           (stacker-state (unwrap! (get-stacker-info stacker) (err ERR_STACK_EXTEND_NOT_LOCKED)))
           (amount-ustx (get locked stacker-info))
           (unlock-height (get unlock-height stacker-info))
+          (current-signing-key (get signing-key stacker-state))
```

Comment:
maybe simplify to
`(next-signing-key (defaults-to (get signing-key stacker-state) updated-signing-key))`

Reply:
Agreed